A number of TypeScript errors in the exported types

### Description

```
node_modules/video.js/dist/types/event-target.d.ts:9:13 - error TS2395: Individual declarations in merged declaration 'EventTarget' must be all exported or all local.

9 export type EventTarget = () => any;
              ~~~~~~~~~~~

node_modules/video.js/dist/types/event-target.d.ts:18:15 - error TS2395: Individual declarations in merged declaration 'EventTarget' must be all exported or all local.

18 declare class EventTarget {
                 ~~~~~~~~~~~

node_modules/video.js/dist/types/plugin.d.ts:5:13 - error TS2395: Individual declarations in merged declaration 'Plugin' must be all exported or all local.

5 export type Plugin = {
              ~~~~~~

node_modules/video.js/dist/types/plugin.d.ts:35:15 - error TS2395: Individual declarations in merged declaration 'Plugin' must be all exported or all local.

35 declare class Plugin {
                 ~~~~~~

node_modules/video.js/dist/types/tech/tech.d.ts:5:13 - error TS2395: Individual declarations in merged declaration 'Tech' must be all exported or all local.

5 export type Tech = any | string;
              ~~~~

node_modules/video.js/dist/types/tech/tech.d.ts:12:15 - error TS2395: Individual declarations in merged declaration 'Tech' must be all exported or all local.

12 declare class Tech extends Component {
                 ~~~~

node_modules/video.js/dist/types/tech/tech.d.ts:239:15 - error TS2552: Cannot find name 'TimeRange'. Did you mean 'TimeRanges'?

239     played(): TimeRange;
                  ~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:22227:13
    22227 declare var TimeRanges: {
                      ~~~~~~~~~~
    'TimeRanges' is declared here.

node_modules/video.js/dist/types/tracks/audio-track-list.d.ts:15:26 - error TS2304: Cannot find name 'AudioTrack'.

15     constructor(tracks?: AudioTrack[]);
                            ~~~~~~~~~~

node_modules/video.js/dist/types/tracks/html-track-element.d.ts:8:15 - error TS2415: Class 'HTMLTrackElement' incorrectly extends base class 'EventTarget'.
  Property 'allowedEvents_' is private in type 'EventTarget' but not in type 'HTMLTrackElement'.

8 declare class HTMLTrackElement extends EventTarget {
                ~~~~~~~~~~~~~~~~

node_modules/video.js/dist/types/tracks/text-track-list.d.ts:17:5 - error TS2416: Property 'addTrack' in type 'TextTrackList' is not assignable to the same property in base type 'TrackList'.
  Type '(track: TextTrack) => void' is not assignable to type '(track: Track) => void'.
    Types of parameters 'track' and 'track' are incompatible.
      Type 'Track' is missing the following properties from type 'TextTrack': tech_, cues_, activeCues_, preload_, and 8 more.

17     addTrack(track: import('./text-track').default): void;
       ~~~~~~~~

node_modules/video.js/dist/types/tracks/text-track.d.ts:8:15 - error TS2415: Class 'TextTrack' incorrectly extends base class 'Track'.
  Property 'allowedEvents_' is private in type 'Track' but not in type 'TextTrack'.

8 declare class TextTrack extends Track {
                ~~~~~~~~~

node_modules/video.js/dist/types/tracks/track-list.d.ts:8:15 - error TS2415: Class 'TrackList' incorrectly extends base class 'EventTarget'.
  Types have separate declarations of a private property 'allowedEvents_'.

8 declare class TrackList extends EventTarget {
                ~~~~~~~~~

node_modules/video.js/dist/types/video.d.ts:88:14 - error TS2304: Cannot find name 'version'.

88     export { version as VERSION };
                ~~~~~~~
```

### Reduced test case

No need for a test case

### Steps to reproduce

Just install `video.js` and run a TypeScript check with this `tsconfig.json`:

```json
{
  "compilerOptions": {
    "target": "ESNext",
    "allowJs": false,
    "allowUnreachableCode": false,
    "allowUnusedLabels": false,
    "forceConsistentCasingInFileNames": true,
    "moduleResolution": "Node",
    "useDefineForClassFields": true,
    "strict": true,
    "exactOptionalPropertyTypes": true,
    "noUncheckedIndexedAccess": true,
    "newLine": "lf",
    "noFallthroughCasesInSwitch": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noImplicitOverride": true,
    "noImplicitReturns": true,
    "noEmitOnError": true,
    "noEmit": true,
    "verbatimModuleSyntax": true,
    "declaration": true,
    "declarationMap": true,
    "inlineSourceMap": true,
    "inlineSources": true,
    "baseUrl": ".",
    "module": "ESNext",
    "experimentalDecorators": true
  }
}
```

The only workaround is adding `"skipLibCheck": true`.

Perhaps related to https://github.com/videojs/video.js/issues/8141.

Happens on TypeScript 5.1.3 but most likely it happens also on older versions.

### Errors

_No response_

### What version of Video.js are you using?

8.3.0

### Video.js plugins used.

_No response_

### What browser(s) including version(s) does this occur with?

N/A

### What OS(es) and version(s) does this occur with?

Linux

--- a/src/js/component.js
+++ b/src/js/component.js
@@ -1076,9 +1076,8 @@ class Component {
    * @param {boolean} [skipListeners]
    *        Skip the componentresize event trigger
    *
-   * @return {number|string}
-   *         The width when getting, zero if there is no width. Can be a string
-   *           postpixed with '%' or 'px'.
+   * @return {number|undefined}
+   *         The width when getting, zero if there is no width
    */
   width(num, skipListeners) {
     return this.dimension('width', num, skipListeners);
@@ -1094,9 +1093,8 @@ class Component {
    * @param {boolean} [skipListeners]
    *        Skip the componentresize event trigger
    *
-   * @return {number|string}
-   *         The width when getting, zero if there is no width. Can be a string
-   *         postpixed with '%' or 'px'.
+   * @return {number|undefined}
+   *         The height when getting, zero if there is no height
    */
   height(num, skipListeners) {
     return this.dimension('height', num, skipListeners);
@@ -1142,7 +1140,7 @@ class Component {
    * @param  {boolean} [skipListeners]
    *         Skip componentresize event trigger
    *
-   * @return {number}
+   * @return {number|undefined}
    *         The dimension when getting or 0 if unset
    */
   dimension(widthOrHeight, num, skipListeners) {

--- a/src/js/event-target.js
+++ b/src/js/event-target.js
@@ -168,7 +168,7 @@ class EventTarget {
 /**
  * All event listeners should follow the following format.
  *
- * @callback EventTarget~EventListener
+ * @callback EventListener
  * @this {EventTarget}
  *
  * @param {Event} event
@@ -185,7 +185,7 @@ class EventTarget {
  *         will have extra functionality. See that function for more information.
  *
  * @property EventTarget.prototype.allowedEvents_
- * @private
+ * @protected
  */
 EventTarget.prototype.allowedEvents_ = {};
 

--- a/src/js/plugin.js
+++ b/src/js/plugin.js
@@ -81,7 +81,7 @@ const markPluginAsActive = (player, name) => {
  * @param  {Player} player
  *         A Video.js player instance.
  *
- * @param  {Plugin~PluginEventHash} hash
+ * @param  {PluginEventHash} hash
  *         A plugin event hash.
  *
  * @param  {boolean} [before]
@@ -238,7 +238,7 @@ class Plugin {
    * @param   {Object} [hash={}]
    *          An object to be used as event an event hash.
    *
-   * @return {Plugin~PluginEventHash}
+   * @return {PluginEventHash}
    *          An event hash object with provided properties mixed-in.
    */
   getEventHash(hash = {}) {
@@ -257,7 +257,7 @@ class Plugin {
    *
    * @param   {Object} [hash={}]
    *          Additional data hash to merge with a
-   *          {@link Plugin~PluginEventHash|PluginEventHash}.
+   *          {@link PluginEventHash|PluginEventHash}.
    *
    * @return {boolean}
    *          Whether or not default was prevented.
@@ -482,7 +482,7 @@ export default Plugin;
  * Signals that a plugin is about to be set up on a player.
  *
  * @event    Player#beforepluginsetup
- * @type     {Plugin~PluginEventHash}
+ * @type     {PluginEventHash}
  */
 
 /**
@@ -490,14 +490,14 @@ export default Plugin;
  * is the name of the plugin.
  *
  * @event    Player#beforepluginsetup:$name
- * @type     {Plugin~PluginEventHash}
+ * @type     {PluginEventHash}
  */
 
 /**
  * Signals that a plugin has just been set up on a player.
  *
  * @event    Player#pluginsetup
- * @type     {Plugin~PluginEventHash}
+ * @type     {PluginEventHash}
  */
 
 /**
@@ -505,11 +505,11 @@ export default Plugin;
  * is the name of the plugin.
  *
  * @event    Player#pluginsetup:$name
- * @type     {Plugin~PluginEventHash}
+ * @type     {PluginEventHash}
  */
 
 /**
- * @typedef  {Object} Plugin~PluginEventHash
+ * @typedef  {Object} PluginEventHash
  *
  * @property {string} instance
  *           For basic plugins, the return value of the plugin function. For

--- a/src/js/tech/tech.js
+++ b/src/js/tech/tech.js
@@ -22,7 +22,7 @@ import * as Guid from '../utils/guid.js';
  * * `var SourceObject = {src: 'http://ex.com/video.mp4', type: 'video/mp4'};`
    * `var SourceString = 'http://example.com/some-video.mp4';`
  *
- * @typedef {Object|string} Tech~SourceObject
+ * @typedef {Object|string} SourceObject
  *
  * @property {string} src
  *           The url to the source
@@ -480,7 +480,7 @@ class Tech extends Component {
    * > NOTE: This implementation is incomplete. It does not track the played `TimeRange`.
    *         It only checks whether the source has played at all or not.
    *
-   * @return {TimeRange}
+   * @return { import('../utils/time').TimeRange }
    *         - A single time range if this video has played
    *         - An empty set of ranges if not.
    */
@@ -1280,7 +1280,7 @@ Tech.withSourceHandlers = function(_Tech) {
    *
    * TODO: Answer question: should 'probably' be prioritized over 'maybe'
    *
-   * @param {Tech~SourceObject} source
+   * @param {SourceObject} source
    *        The source object
    *
    * @param {Object} options
@@ -1308,7 +1308,7 @@ Tech.withSourceHandlers = function(_Tech) {
   /**
    * Check if the tech can support the given source.
    *
-   * @param {Tech~SourceObject} srcObj
+   * @param {SourceObject} srcObj
    *        The source object
    *
    * @param {Object} options
@@ -1371,7 +1371,7 @@ Tech.withSourceHandlers = function(_Tech) {
    * and source handlers.
    * Should never be called unless a source handler was found.
    *
-   * @param {Tech~SourceObject} source
+   * @param {SourceObject} source
    *        A source object with src and type keys
    */
   _Tech.prototype.setSource = function(source) {

--- a/src/js/tracks/audio-track-list.js
+++ b/src/js/tracks/audio-track-list.js
@@ -36,7 +36,7 @@ class AudioTrackList extends TrackList {
   /**
    * Create an instance of this class.
    *
-   * @param {AudioTrack[]} [tracks=[]]
+   * @param { import('./audio-track').default[] } [tracks=[]]
    *        A list of `AudioTrack` to instantiate the list with.
    */
   constructor(tracks = []) {

--- a/src/js/tracks/html-track-element.js
+++ b/src/js/tracks/html-track-element.js
@@ -105,6 +105,9 @@ class HTMLTrackElement extends EventTarget {
   }
 }
 
+/**
+ * @protected
+ */
 HTMLTrackElement.prototype.allowedEvents_ = {
   load: 'load'
 };

--- a/src/js/tracks/text-track.js
+++ b/src/js/tracks/text-track.js
@@ -440,6 +440,7 @@ class TextTrack extends Track {
 
 /**
  * cuechange - One or more cues in the track have become active or stopped being active.
+ * @protected
  */
 TextTrack.prototype.allowedEvents_ = {
   cuechange: 'cuechange'

--- a/src/js/tracks/track-list.js
+++ b/src/js/tracks/track-list.js
@@ -177,7 +177,7 @@ class TrackList extends EventTarget {
  * Events that can be called with on + eventName. See {@link EventHandler}.
  *
  * @property {Object} TrackList#allowedEvents_
- * @private
+ * @protected
  */
 TrackList.prototype.allowedEvents_ = {
   change: 'change',

--- a/src/js/video.js
+++ b/src/js/video.js
@@ -2,6 +2,9 @@
  * @file video.js
  * @module videojs
  */
+/**
+ * @typedef { string } version
+ */
 import {version} from '../../package.json';
 import window from 'global/window';
 import {

